### PR TITLE
Add support for Jenkins Blue Ocean

### LINKED ISSUE #55: Add support for Jenkins Blue Ocean
Blue Ocean supports pull requests directly and uses a new version of the SCM plugin. This new version has changed the following environment variables used by codecov:

`GIT_BRANCH` has been replaced with `BRANCH_NAME`
`GIT_COMMIT` has no direct replacement (accessible only with Groovy)

SInce Blue Ocean no longer uses the GitHub Pull Request Builder plugin, all `ghprb` prefixed variables are not longer defined either, so:

`ghrpbSourceBranch` has been replaced with `BRANCH_NAME`
`ghrpbActualCommit` has no direct replacement (accessible only with Groovy)
`ghprbPullId` has been replaced with `CHANGE_ID`

The other environment variables are still available as before.

Here is a working Groovy implementation from the Docker team:
https://github.com/docker/jenkins-pipeline-scripts/blob/master/vars/codecov.groovy

References:
https://jenkins.io/projects/blueocean/
https://jenkins.io/doc/book/pipeline/multibranch/#additional-environment-variables

### PR DESCRIPTION
**Pull request by rochdev**
closes #55

### CODE REVIEW

**Comment by stevepeak** [comment]
Very nice job  @rochdev Thank you!

File: lib/git.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+var execSync = require("child_process").execSync;
+
+module.exports = {
+
+  branch: function(){
+    return execSync("git rev-parse --abbrev-ref HEAD || hg branch").toString().trim();
+  },
+
+  head: function(){
+    return execSync("git rev-parse HEAD || hg id -i --debug | tr -d '+'").toString().trim();
```
Comment (stevepeak) [inline lib/git.js]: Recommended change:

```diff
- return execSync("git rev-parse HEAD || hg id -i --debug | tr -d '+'").toString().trim()
+ return execSync("git log -1 --pretty=%B || hg id -i --debug | tr -d '+'").toString().trim()
```

Comment (rochdev) [inline lib/git.js]: @stevepeak This change seems to return the commit message instead of the commit ID.

**Comment by rochdev** [comment]
Should I add Blue Ocean to the README? It might be confusing if different codecov implementations have different levels of support for Jenkins that is not documented.

**Comment by eddiemoore** [comment]
Looks good @rochdev 

@stevepeak Do the other uploaders support Blue Ocean? If not we can add it to them

**Comment by rochdev** [comment]
@stevepeak @eddiemoore What is the next step to have this move forward?

**Comment by stevepeak** [comment]
@eddiemoore Other uploaders should support Blue Ocean by using `git / hg`.

**Comment by stevepeak** [comment]
Given the change to `git log -1 --format="%H"`: LGTM

**Comment by eddiemoore** [comment]
Looks good to me. @stevepeak All ok with you?

**Comment by stevepeak** [comment]
Yes, thank you @eddiemoore.

> We can do `git rev-parse HEAD` to `git log -1 --format="%H"` after merge.

**Pull request merged**